packages/admin-ui/src/menu/menu-item.tsx

```diff
@@ -1,46 +1,49 @@
-import type { ReactNode } from 'react'
-import React from 'react'
-import {
-  createComponent,
-  useElement,
-  IconContainer,
-} from '@vtex/admin-ui-react'
+import type { ReactNode, Ref } from 'react'
+import React, { forwardRef } from 'react'
+import { IconContainer } from '@vtex/admin-ui-react'
 import { MenuItem as AriakitMenuItem } from 'ariakit/menu'
 
 import { MenuItemWrapper } from './menu-item-wrapper'
 import { Center } from '../center'
 
-import * as style from './menu.style'
+import { itemTheme } from './menu.css'
+import { cx } from '@vtex/admin-ui-core'
 
-export const MenuItem = createComponent<
-  typeof AriakitMenuItem,
-  MenuItemOptions
->((props) => {
-  const { icon, label, critical, disabled, onClick, ...itemProps } = props
-
-  return useElement(MenuItemWrapper, {
-    disabled,
-    children: useElement(AriakitMenuItem, {
+export const MenuItem = forwardRef(
+  (props: MenuItemProps, ref: Ref<HTMLDivElement>) => {
+    const {
+      icon,
+      label,
+      critical,
       disabled,
       onClick,
-      ...itemProps,
-      baseStyle: {
-        ...style.item,
-        ...style.itemVariants({
-          variant:
-            (disabled && 'disabled') || (critical && 'critical') || 'neutral',
-        }),
-      },
-      children: (
-        <Center>
-          <IconContainer size="small">{icon}</IconContainer> {label}
-        </Center>
-      ),
-    }),
-  })
-})
+      className = '',
+      ...itemProps
+    } = props
+
+    const variant =
+      (disabled && 'disabled') || (critical && 'critical') || 'neutral'
 
-export type MenuItemOptions = {
+    return (
+      <MenuItemWrapper disabled={disabled}>
+        <AriakitMenuItem
+          disabled={disabled}
+          ref={ref}
```

Comment:
Unsure if this ref should be here on in wrapper. Same for classname.

Both menuwrapper and menuItem have classnames that someone might want to overwrite

Reply:
It should be on the wrapper

Reply:
I just saw that the wrapper can be a fragment, this way it makes sense to both ref and className to be part of AriakitMenuItem